build/dsc/manager.rb

```diff
@@ -168,13 +168,21 @@ def document_types(markdown_file_path, dsc_types)
         file.write("For any system this module is installed on, use\n`Puppet describe typename` for more information.\n\n")
 
         file.write("#### WMF Core Types\n\n")
-        dsc_types.select { |t| t.match(/dsc_[^x]/)}.each do |dsc_type|
-          file.write("* #{dsc_type}\n")
+        file.write("Puppet Type | DSC Resource Name\n")
+        file.write("----------- | -----------------\n")
+        resources.select { |t| t.instance_name.downcase.match(/dsc_[^x]/)}.each do |dsc_type|
+          file.write("dsc_#{dsc_type.friendlyname.downcase} | #{dsc_type.friendlyname}\n")
         end
 
         file.write("\n#### Community x Prefixed types\n\n")
-        dsc_types.select { |t| t.match(/dsc_x/)}.each do |dsc_type|
-          file.write("* #{dsc_type}\n")
+        resources.select { |t| t.instance_name.downcase.match(/dsc_x/)}.group_by { |dt| dt.ps_module.name}.each do |dsc_type|
+          file.write("##### #{dsc_type[0]}\n\n")
+          file.write("Puppet Type | DSC Resource Name\n")
+          file.write("----------- | -----------------\n")
+          dsc_type[1].each do | sb |
+            file.write("dsc_#{sb.friendlyname.downcase} | #{sb.friendlyname}\n")
```

Comment:
I chose this order to show Puppet types first, but should this be DSC Resource then Puppet type?